### PR TITLE
Pressure inlet boundary condition in the Navier Stokes FV action

### LINKED ISSUE #21380: Inlet pressure option in the Navier-Stokes FV action
## Reason

Some problems require inlet pressure boundaries AND the ability to specify the inlet
temperature.

## Design

Pretty much copy-paste the outlet pressure implementation and minor modifications

## Impact

Added option in the NSFV action. No interference with existing capability are expected.

@grmnptr

### PR DESCRIPTION
**Pull request by snschune**
closes #21380 

## Reason

Chimney like natural convection problems require pressure inlet conditions. Other problems
might need that too.

## Design

Small changes to the NSFV action. 

## Impact

The input for the momentum inlet function had to change slightly
to accommodate pressure functions.

@grmnptr

### CODE REVIEW

File: modules/navier_stokes/src/actions/NSFVAction.C [hunk 1]
```diff
@@ -2370,20 +2392,45 @@ NSFVAction::checkBoundaryParameterErrors()
...
-                 "direction dependent directions",
-                 "'inlet_boundaries' times the mesh dimension");
+                 num_dir_dependent_bcs + num_pressure_inlet_bcs,
+                 "fixed-velocity and fixed-pressure entries",
+                 "momentum_inlet_types");
+
+  unsigned int k = 0;
+  for (unsigned int j = 0; j < _momentum_inlet_types.size(); ++j)
+  {
+    const auto & type = _momentum_inlet_types[j];
```
Comment (lindsayad) [inline modules/navier_stokes/src/actions/NSFVAction.C:2416]:
```suggestion
  for (const auto & type : _momentum_inlet_types)
  {
```

File: modules/navier_stokes/src/actions/NSFVAction.C [hunk 2]
```diff
@@ -2370,20 +2392,45 @@ NSFVAction::checkBoundaryParameterErrors()
...
   checkSizeParam(_momentum_inlet_function.size(),
                  "momentum_inlet_function",
-                 num_dir_dependent_bcs * _dim,
-                 "direction dependent directions",
-                 "'inlet_boundaries' times the mesh dimension");
+                 num_dir_dependent_bcs + num_pressure_inlet_bcs,
+                 "fixed-velocity and fixed-pressure entries",
+                 "momentum_inlet_types");
+
+  unsigned int k = 0;
```
Comment (lindsayad) [inline modules/navier_stokes/src/actions/NSFVAction.C:2413]:
```suggestion
  // index into _momentum_inlet_function
  unsigned int k = 0;
```

File: modules/navier_stokes/test/tests/finite_volume/ins/action/errors/tests [hunk 3]
```diff
@@ -12,9 +12,17 @@
...
+    requirement = 'The system shall throw an error if the number of momentum inlet functions is not equal to the problem dimension for a fixed-velocity inlet.'
     cli_args = "Modules/NavierStokesFV/momentum_inlet_function='1'"
-    expect_err = "Size \(1\) is not the same as the number of direction dependent directions in 'inlet_boundaries' times the mesh dimension \(2\)"
+    expect_err = "Size \(1\) is not the same as the number of entries  in  the momentum_inlet_types subvector for fixed-velocity \(2\)"
+    ad_indexing_type = 'global'
+  []
+  [momentum-inlet-function-error-2]
+    type = 'RunException'
+    input = 2d-rc-error-action.i
+    requirement = 'The system shall throw an error if the number of entries in momentum_inlet_types does not match the total number of fixed-velocity and fixed-pressure inlet boundaries.'
```
Comment (lindsayad) [inline modules/navier_stokes/test/tests/finite_volume/ins/action/errors/tests:23]: We've been told that requirement language should be as free of implementation details as possible. I think here you can just do
```suggestion
    requirement = 'The system shall throw an error if the number of entries for momentum inlet types does not match the total number of fixed-velocity and fixed-pressure inlet boundaries.'
```

**Review by lindsayad — CHANGES_REQUESTED** [review]
I recently tried a pairing of inlet and outlet pressure boundary conditions and failed, so it's nice to see that you were able to drum up a test case

**Comment by snschune** [comment]
@lindsayad I made that test a verification exercise and it converges to the hand calculation.

File: modules/navier_stokes/src/actions/NSFVAction.C [hunk 4]
```diff
@@ -2370,20 +2392,45 @@ NSFVAction::checkBoundaryParameterErrors()
...
+    const auto & type = _momentum_inlet_types[j];
+    if (type != "fixed-velocity" && type != "fixed-pressure")
+      continue;
+
+    if (type == "fixed-velocity")
+      checkSizeParam(_momentum_inlet_function[k].size(),
+                     "momentum_inlet_function",
+                     _dim,
+                     "entries ",
+                     " the momentum_inlet_types subvector for fixed-velocity");
```
Comment (GiudGiud) [inline modules/navier_stokes/src/actions/NSFVAction.C:2425]:
```suggestion
                     " the momentum_inlet_types subvector for fixed-velocity on inlet: " + _inlet_boundaries[k]);
```

File: modules/navier_stokes/src/actions/NSFVAction.C [hunk 5]
```diff
@@ -2370,20 +2392,45 @@ NSFVAction::checkBoundaryParameterErrors()
...
+    else if (type == "fixed-pressure")
+      checkSizeParam(_momentum_inlet_function[k].size(),
+                     "momentum_inlet_function",
+                     1,
+                     "entries ",
+                     " the momentum_inlet_types subvector for fixed-pressure");
```
Comment (GiudGiud) [inline modules/navier_stokes/src/actions/NSFVAction.C:2431]:
```suggestion
                     " the momentum_inlet_types subvector for fixed-pressure inlet : " + _inlet_boundaries[k]);
```

**Comment by GiudGiud** [comment]
Seems the new test is diffing in parallel on integral quantities

**Comment by grmnptr** [comment]
Okay, turned out to be a tolerance issue. Should be good now.

**Review by GiudGiud — APPROVED** [review]

**Review by lindsayad — APPROVED** [review]

**Pull request merged**